### PR TITLE
Added sleep between upload retries.

### PR DESCRIPTION
**Pull request by nmoinvaz**
See #293.

### CODE REVIEW

**Comment by nmoinvaz** [comment]
Not sure why the failure on `black` on a single instance. I checked manually and that file `tests\test.py` appears fine.

**Comment by nmoinvaz** [comment]
This PR:

* Adds sleep 2s for each retry (same as codecov bash uploader)
* Sets default retries to 5 (same as codecov bash uploader)
* Logs retry attempts to output for debugging.

File: codecov/__init__.py [hunk 1]
```diff
@@ -287,11 +287,14 @@ def generate_toc(root):
...
 
-def retry_upload(url, request_method, retries=3, break_codes=(200,), **kwargs):
-    for _ in range(retries):
+def retry_upload(url, request_method, retries=5, break_codes=(200,), **kwargs):
+    wait_seconds = 2
+    for i in range(retries):
         res = request_method(url, **kwargs)
         if res.status_code in break_codes:
             return res
+        write("    Retrying {0}/{1} in {2}s..".format(i+1, retries, wait_seconds))
```
Comment (ktbarrett) [inline codecov/__init__.py:296]:
```suggestion
        write("    Retrying {0}/{1} in {2}s..".format(i + 1, retries, wait_seconds))
```
Fixes the CI error.

**Comment by nmoinvaz** [comment]
I have fixed the CI issue.

**Review by thomasrockhu — APPROVED** [review]

**Pull request merged**